runtime/lua/provider/perl/health.lua

```diff
@@ -9,19 +9,14 @@ function M.check()
     return
   end
 
-  local perl_detect_table = vim.fn['provider#perl#Detect']()
-  local perl_exec = perl_detect_table[1]
-  local perl_warnings = perl_detect_table[2]
+  local perl_exec, perl_warnings = require('vim.provider.perl').detect()
 
-  if perl_exec:find('^%s*$') then
-    if perl_warnings:find('%S') then
-      health.warn(perl_warnings, {
-        'See :help provider-perl for more information.',
-        'You may disable this provider (and warning) by adding `let g:loaded_perl_provider = 0` to your init.vim',
-      })
-    else
-      health.warn('No usable perl executable found')
-    end
+  if not perl_exec then
+    health.warn(assert(perl_warnings), {
+      'See :help provider-perl for more information.',
+      'You may disable this provider (and warning) by adding `let g:loaded_perl_provider = 0` to your init.vim',
+    })
+    health.warn('No usable perl executable found')
```

Comment:
When user set an invalid `g:perl_host_prog`, it doesn't speak anything about the wrong `g:perl_host_prog`. Like in python provider helathcheck, it'd be good to print out `g:perl_host_prog` if set.

Reply:
Yeah, I think I'd prefer to just dump the code over to the lua side as is and treat any logical improvements as out of scope for the "lua-ify" PRs.